engine/src/flutter/display_list/dl_paint.h

```diff
@@ -123,57 +123,77 @@ class DlPaint {
     return *this;
   }
 
-  std::shared_ptr<const DlColorSource> getColorSource() const {
+  const std::shared_ptr<const DlColorSource>& getColorSource() const {
```

Comment:
return a ref so that callers that specify a const ref don't increment/decrement.

Reply:
Isn't this sufficient alone to get rid of all those increment decrements, so the getColorSourcePtr calls aren't required?

Reply:
in theory, yes, but since we were calling `getColorSource().get()` anyway its still an improvement imo

Reply:
Yeah, get().get() should be outlawed.